openedx/features/content_type_gating/models.py

```diff
@@ -109,18 +106,7 @@ def enabled_for_enrollment(cls, enrollment=None, user=None, course_key=None):
                 return False
 
         # check if user is in holdback
-        is_in_holdback = False
-        if user and user.is_authenticated and (user_variable_represents_correct_user):
-            try:
-                holdback_value = ExperimentData.objects.get(
-                    user=user,
-                    experiment_id=EXPERIMENT_ID,
-                    key=EXPERIMENT_DATA_HOLDBACK_KEY,
-                ).value
-                is_in_holdback = holdback_value == 'True'
-            except ExperimentData.DoesNotExist:
-                pass
-        if is_in_holdback:
+        if user_variable_represents_correct_user and cls._is_in_holdback(user):
```

Comment:
how are these changes related to track selection?  
I would just do some local testing or a local demo around the holdback if we change that code, since i think the automated tests around it are limited

Reply:
This code was already checking for holdback, but doing so after checking for enrollment. I changed the order to first check for holdback (and to use util method), so it mimics what was done in CourseDurationLimitConfig (see this PR which landed https://github.com/edx/edx-platform/pull/19461).
I believe it's necessary because we are now checking both models when determining what to show on the track selection page (https://github.com/edx/edx-platform/pull/19470/files#diff-d770ce009be697aedec075798accf1ceR196).

I did a bunch of manual testing of different scenarios and all appears to be working correctly.
Let me know if I should be writing additional automated tests around this area.

Reply:
if your manual testing included verifying that the holdback bucketing specifically was working correctly, i think that's fine